crates/sui-benchmark/src/bin/stress.rs

```diff
@@ -0,0 +1,543 @@
+                                }
+                            }
+                        }
+                    }
+                }
+            });
+            tasks.push(runner);
+        });
+
+    tasks.push(tokio::spawn(async move {
+            let mut stat_collection: BTreeMap<usize, Stats> = BTreeMap::new();
+            while let Some(s @ Stats {
+                id,
+                num_success: _,
+                num_error: _,
+                min_latency: _,
+                max_latency: _,
+                num_no_gas: _,
+                num_in_flight: _,
+                num_submitted: _,
+                duration
+            }) = rx.recv().await {
+                stat_collection.insert(id, s);
+                let mut total_qps: f32 = 0.0;
+                let mut num_success: u64 = 0;
+                let mut num_error: u64 = 0;
+                let mut min_latency: Duration = Duration::MAX;
+                let mut max_latency: Duration = Duration::ZERO;
+                let mut num_in_flight: u64 = 0;
+                let mut num_submitted: u64 = 0;
+                let mut num_no_gas = 0;
+                for (_, v) in stat_collection.iter() {
+                    total_qps += v.num_success as f32 / duration.as_secs() as f32;
+                    num_success += v.num_success;
+                    num_error += v.num_error;
+                    num_no_gas += v.num_no_gas;
+                    num_submitted += v.num_submitted;
+                    num_in_flight += v.num_in_flight;
+                    min_latency = if v.min_latency < min_latency {
+                        v.min_latency
+                    } else {
+                        min_latency
+                    };
+                    max_latency = if v.max_latency > max_latency {
+                        v.max_latency
+                    } else {
+                        max_latency
+                    };
+                }
+                let denom = num_success + num_error;
+                let _error_rate = if denom > 0 {
+                    num_error as f32 / denom as f32
+                } else {
+                    0.0
+                };
+                write(format!("Throughput = {}, min_latency_ms = {}, max_latency_ms = {}, num_success = {}, num_error = {}, no_gas = {}, submitted = {}, in_flight = {}", total_qps, min_latency.as_millis(), max_latency.as_millis(), num_success, num_error, num_no_gas, num_submitted, num_in_flight));
+            }
+        }));
+    let _: Vec<_> = try_join_all(tasks).await.unwrap().into_iter().collect();
+}
+
+#[tokio::main]
+async fn main() {
+    let env_filter = EnvFilter::try_from_default_env().unwrap_or_else(|_| EnvFilter::new("warn"));
+    let subscriber_builder =
+        tracing_subscriber::fmt::Subscriber::builder().with_env_filter(env_filter);
+    let subscriber = subscriber_builder.with_writer(std::io::stderr).finish();
+    set_global_default(subscriber).expect("Failed to set subscriber");
```

Comment:
You should use the telemetry-subscriber crate here like we do in our other binaries.

Reply:
done